### PR TITLE
fixing resources reporting as updated

### PR DESCRIPTION
**Pull request by jeremymv2**
### Description

This PR removes the dependency on `chef_chandler` because that cookbook's LWRP reports updated resources for each handler enabled.  It also removes the `inspec` custom resource and simply makes it a `chef_gem` resource with `compile_time true`.  This makes the assumption that inspec gem version is ALWAYS specified via attribute to a specific SEMVER version (`node['audit']['inspec_version']`)  - "latest" is no longer allowed.  

### Issues Resolved

This addresses #138 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

**Comment by jeremymv2** [comment]
The result of testing collector = chef-server

Notice 0/x resources updated.

```
root@node:/tmp/vagrant-chef# chef-client -c client.rb
[2016-11-02T22:46:54+00:00] INFO: Forking chef instance to converge...
Starting Chef Client, version 12.15.19
[2016-11-02T22:46:54+00:00] INFO: *** Chef 12.15.19 ***
[2016-11-02T22:46:54+00:00] INFO: Platform: x86_64-linux
[2016-11-02T22:46:54+00:00] INFO: Chef-client pid: 2889
[2016-11-02T22:46:56+00:00] INFO: Run List is [recipe[audit_wrapper::default]]
[2016-11-02T22:46:56+00:00] INFO: Run List expands to [audit_wrapper::default]
[2016-11-02T22:46:56+00:00] INFO: Starting Chef Run for node
[2016-11-02T22:46:56+00:00] INFO: Running start handlers
[2016-11-02T22:46:56+00:00] INFO: Start handlers complete.
[2016-11-02T22:46:56+00:00] INFO: HTTP Request Returned 404 Not Found:
resolving cookbooks for run list: ["audit_wrapper::default"]
[2016-11-02T22:46:56+00:00] INFO: Loading cookbooks [audit_wrapper@0.1.0, audit@2.0.0, compat_resource@12.16.1]
Synchronizing Cookbooks:
  - audit_wrapper (0.1.0)
  - audit (2.0.0)
  - compat_resource (12.16.1)
Installing Cookbook Gems:
Compiling Cookbooks...
Recipe: audit::default
  * chef_gem[inspec] action install (up to date)
[2016-11-02T22:46:56+00:00] INFO: loading handler from /var/chef/cache/cookbooks/audit/files/default/handler/audit_report
  Converging 1 resources
  * chef_gem[inspec] action install (up to date)
[2016-11-02T22:46:56+00:00] INFO: Chef Run complete in 0.283401384 seconds

Running handlers:
[2016-11-02T22:46:56+00:00] INFO: Running report handlers
[2016-11-02T22:46:56+00:00] WARN: Format is json-min
[2016-11-02T22:46:56+00:00] INFO: Initialize InSpec
[2016-11-02T22:46:56+00:00] INFO: Running tests from: [{:name=>"linux", :compliance=>"base/linux"}]
[2016-11-02T22:46:57+00:00] INFO: Reporting to chef-server
[2016-11-02T22:46:57+00:00] INFO: Control Profile: ["linux"]
[2016-11-02T22:46:57+00:00] INFO: Control Profil: linux
[2016-11-02T22:46:57+00:00] INFO: Compliance Profils: [{:owner=>"base", :profile_id=>"linux"}]
[2016-11-02T22:46:57+00:00] INFO: Report to Chef Server: https://chef-server.test/compliance/organizations/brewinc/inspec
  - Chef::Handler::AuditReport
Running handlers complete
[2016-11-02T22:46:57+00:00] INFO: Report handlers complete
Chef Client finished, 0/2 resources updated in 02 seconds
root@node:/tmp/vagrant-chef#
```

File: libraries/helper.rb [hunk 1]
```diff
@@ -106,4 +106,18 @@ def handle_reporters(reporters)
   def cookbook_vendor_path
     File.expand_path('../../files/default/vendor', __FILE__)
   end
+
+  def cookbook_handler_path
+    File.expand_path('../../files/default/handler', __FILE__)
+  end
+
+  def load_audit_handler
+    libpath = ::File.join(cookbook_handler_path, 'audit_report')
```
Comment (chris-rock) [inline libraries/helper.rb]: Very nice approach!

File: recipes/default.rb [hunk 2]
```diff
@@ -1,22 +1,8 @@
-include_recipe 'chef_handler'
-
-# install inspec
-inspec 'inspec' do
+chef_gem 'inspec' do
   version node['audit']['inspec_version']
```
Comment (chris-rock) [inline recipes/default.rb]: Any reason why we are not using `version new_resource.version if new_resource.version != 'latest'`?

File: metadata.rb [hunk 3]
```diff
@@ -13,4 +13,3 @@
 chef_version '>= 12.5.1' if respond_to?(:chef_version)
 
 depends 'compat_resource'
```
Comment (chris-rock) [inline metadata.rb]: @lamont-granquist Is there any real reason why we need `compat_resource `

**Comment by jeremymv2** [comment]
This now supports `node['audit']['inspec_version'] = 'latest'`

File: resources/inspec.rb [hunk 4]
```diff
@@ -1,34 +0,0 @@
...
-
-  converge_by 'verifies the inspec version' do
-    verify_inspec_version version
-  end
-end
-
-def verify_inspec_version(inspec_version)
-  require 'inspec'
-  # check that we have the right inspec version
-  if Inspec::VERSION != inspec_version && inspec_version !='latest'
```
Comment (chris-rock) [inline resources/inspec.rb]: @jeremymv2 Do you think we should keep that warning. May be very helpful for debugging?

Comment (jeremymv2) [inline resources/inspec.rb]: Good idea @chris-rock 
I've moved it under `load_inspec_libs` function which seemed appropriate :)

File: libraries/compliance.rb [hunk 5]
```diff
@@ -3,6 +3,12 @@
 # load all the inspec and compliance bundle requirements
 def load_inspec_libs
   require 'inspec'
+  if Inspec::VERSION != node['audit']['inspec_version'] && node['audit']['inspec_version'] !='latest'
```
Comment (chris-rock) [inline libraries/compliance.rb]: That is a very nice idea

**Comment by chris-rock** [comment]
Awesome work @jeremymv2

**Pull request merged**